File: ios/Runner/Interface/NativeChannel.swift
//
//  NativeChannel.swift
//  Runner
//
//  Created by 吉田周平 on 2022/11/26.
//

import Foundation
import Flutter

class NativeChannel {
    
    let channel: FlutterMethodChannel
    let plugin: NativePlugin
    
    init(messenger: FlutterBinaryMessenger) {
        channel = FlutterMethodChannel(name: "native-plugin", binaryMessenger: messenger)
        plugin = NativePlugin()
    }
    
    func setCallHandler() {
        channel.setMethodCallHandler ({
            (call: FlutterMethodCall, result: @escaping FlutterResult) -> Void in
            switch call.method {
            case "pose-fn":
                let input = PoseInput(args: call.arguments)
                self.plugin.poseDetect(poseInput: input,
                                       detectComleted: { resultData in
                    result(resultData)
                })
            default:
                result(FlutterMethodNotImplemented)
                return
            }
        })
    }
}

File: android/app/src/main/kotlin/com/example/shus_flutter/common/channel/NativeChannel.kt
package com.example.shus_flutter.common.channel

import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel

class NativeChannel(messenger: BinaryMessenger) {
    private val channel = MethodChannel(messenger, "native-plugin")
    private val plugin = NativePlugin()

    fun setCallHandler() {
        channel.setMethodCallHandler { call, result ->
            when (call.method) {
                "pose-fn" -> {
                    plugin.poseDetect(call.arguments)
                }
            }
        }
    }
}